### PR TITLE
DOC: spatial.distance: update formula for {s,sq}euclidean

### LINKED ISSUE #18271: Broken or wrong formulas on distance definition
### Issue with current documentation:

Broken formula: https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.sqeuclidean.html#scipy.spatial.distance.sqeuclidean
Missing firmula: https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.seuclidean.html#scipy.spatial.distance.seuclidean

### Idea or request for content:

The next page "pdist" offers these same formulas: https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.pdist.html 

### Additional context (e.g. screenshots, GIFs)

_No response_

### PR DESCRIPTION
**Pull request by nobelwong99**
- Updated sqeuclidean equation in definition
- Updated seuclidean summary and added equation in definition

#### Reference issue
Closes: https://github.com/scipy/scipy/issues/18271

#### What does this implement/fix?
This fix updates two function descriptions. 
1. Updated equation in sqeuclidean
2. Updated function summary and added equation in seuclidean

### CODE REVIEW

**Review by tupui — COMMENTED** [review]
Thank you for the PR @nobelwong99!

In reference to the linter and other CI. For doc only changes, you can at least always add `[skip actions] [skip cirrus]`. CircleCI is building the doc and 2 jobs in Azure are doing some lint checks.

File: scipy/spatial/distance.py [hunk 1]
```diff
@@ -888,8 +886,12 @@ def seuclidean(u, v, V):
     """
     Return the standardized Euclidean distance between two 1-D arrays.
 
-    The standardized Euclidean distance between `u` and `v`.
+    Computes the standardized Euclidean distance. The standardized Euclidean distance between two n-vectors u and v is
```
Comment (tupui) [inline scipy/spatial/distance.py:889]: Besides the line length, input parameters need to have single back ticks.
```suggestion
    Computes the standardized Euclidean distance. The standardized
    Euclidean distance between two n-vectors `u` and `v` is
```

Comment (tupui) [reply scipy/spatial/distance.py:889]: Sorry for not being clear, it has to go in the commit message.

e.g. if you accept all the suggestions I made, you can go in the `Files changed` tab, batch all suggestions and when clicking on the commit button you will be able to change the message. Here write a proper message and you can add the skip marks.

File: scipy/spatial/distance.py [hunk 2]
```diff
@@ -544,9 +544,7 @@ def sqeuclidean(u, v, w=None):
 
     .. math::
 
-       {\\|u-v\\|}_2^2
-
-       \\left(\\sum{(w_i |(u_i - v_i)|^2)}\\right)
+       \\sum\\limits_i w_i {||u-v||}_2^2
```
Comment (peterbell10) [inline scipy/spatial/distance.py:547]: What happened to the i subscripts for u and v?

File: scipy/spatial/distance.py [hunk 3]
```diff
@@ -888,7 +886,15 @@ def seuclidean(u, v, V):
     """
     Return the standardized Euclidean distance between two 1-D arrays.
 
-    The standardized Euclidean distance between `u` and `v`.
+    Computes the standardized Euclidean distance. The standardized
```
Comment (peterbell10) [inline scipy/spatial/distance.py:889]: This sentence is redundant. "Returns x. Computes x .

File: scipy/spatial/distance.py (hunk 2 above)
Comment (j-bowhay) [reply scipy/spatial/distance.py:547]: Agree I prefer Warren's suggestion of `\sum_i{w_i |u_i - v_i|^2}`

File: scipy/spatial/distance.py (hunk 3 above)
Comment (nobelwong99) [reply scipy/spatial/distance.py]: I have removed the redundant sentence. Thank you for the feedback.

File: scipy/spatial/distance.py (hunk 2 above)
Comment (nobelwong99) [reply scipy/spatial/distance.py]: I have updated the equation to use Warren's suggestion and added back in the subscripts for u and v. Thanks for the feedback!

**Review by tupui — APPROVED** [review]
LGTM and I checked the rendered output. Thank you again @nobelwong99 et al. for the reviews.

**Comment by nobelwong99** [comment]
Thank you everyone for the review and feedback!

**Pull request merged**